internal/porchutil/function.go

```diff
@@ -0,0 +1,155 @@
+package porchutil
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/GoogleContainerTools/kpt/porch/api/porch/v1alpha1"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/client-go/kubernetes"
+
+	// Enable the GCP Authentication plugin
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const expirationNanoSecond = 3
+
+var (
+	functionAPI     = "/apis/" + v1alpha1.SchemeGroupVersion.String() + "/functions"
+	PorchExpiration = time.Duration(expirationNanoSecond) * time.Second
+)
+
+// NewPorchFunctionClient provides the PorchFunctionClient which connects to Porch server
+// and gets the functions.
+func NewPorchFunctionClient() *PorchFunctionClient {
+	functions, err := visitFunctions()
```

Comment:
nit: visitFunctions signals (to me) that the function is iterating over an existing data, rather than fetching it. consider whether another name would convey the purpose more clearly

Reply:
Sounds good. Yeah, I actually did some code refactoring before uploading but didn't rethink about the function names.